fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+                else:
+                    pass
+            else:
+                pass
+        return _found_stub_name, _found_stub_object
+
+    def _gRPC_Connect_Server(self, auth_type=None, channel=None, stub_class=None, certificate_path=None):
+        """
+
+        :param auth_type: Input Secure Connection type
+        :param channel: gRPC Channel
+        :param stub_class: a Stub Class object from imported pb2 library
+        :param certificate_path: path of the ssl/tls certificate
+        :return: stub class object
+        """
+        # Checking supplied Input Arguments
+        assert channel is not None, "channel argument should not be None type"
+        assert stub_class is not None, "stub class object should not be None type"
+        log = logging.getLogger(__name__)
+        if auth_type.lower() not in ['ssl', 'tls', 'oauth2'] and auth_type.lower().strip() == "none":
+            log.info("gRPC Channel Connection is not secure.")
+            channel_object = grpc.insecure_channel(channel)
+            stub_class_obj = stub_class(channel_object)
+            return stub_class_obj, channel_object
+        elif auth_type.lower().find('ssl') != -1 or auth_type.lower().find('tls') != -1:
+            log.info("Channel Authentication is secure : {}".format(auth_type))
+            assert certificate_path is not None, "Please specify the valid certificate path"
+            assert certificate_path.lower() != 'none', "Please specify the valid certificate path"
+            with open(certificate_path, 'rb') as certificate:
+                channel_cred = grpc.ssl_channel_credentials(root_certificates=certificate.read())
+            channel_object = grpc.secure_channel(channel, channel_cred)
+            stub_class_obj = stub_class(channel_object)
+            return stub_class_obj, channel_object
+        else:
+            raise NotImplementedError("Not Implemented these {} Authentication types ".format(auth_type))
+
+    @function("function_grpc")
+    def _function_grpc_function(self, event, *args, **kwargs):
+        """Function: A function to communicate with GRPC Servers based on the Communication Methods Specified."""
+        try:
+            # Get the function parameters:
+            grpc_channel = kwargs.get("grpc_channel")  # text
+            grpc_function = kwargs.get("grpc_function")  # text
+            grpc_function_data = kwargs.get("grpc_function_data")  # text
+
+            log = logging.getLogger(__name__)
+            log.info("grpc_channel: %s", grpc_channel)
+            log.info("grpc_function: %s", grpc_function)
+            log.info("grpc_function_data: %s", grpc_function_data)
+
+            yield StatusMessage("starting...")
+            response_received = None
+            grpc_function_json_data = None
+
+            # Parsing the variable data from config file and resilient inputs
+            try:
+                _grpc_package_name = grpc_function.split(':')[0].strip()
```

Comment:
* Line 191-194 here we are manipulating the function input `grpc_function` which has to been in a certain string format with multiple `:`

* Therefore this code could fail if the user does not enter `grpc_function` in the correct format

* We should write some logic to validate that `grpc_function` is in the correct format and raise a ValueError ro inform the user if the input is incorrect

Reply:
ok